### PR TITLE
Update logic writeup

### LINKED ISSUE #34: Update UML Diagrams in DG to reflect accurate flow of diagram
UML Diagram for the following
- [ ] Add
- [ ] Delete
- [ ] Edit
- [ ] Find
- [ ] Undo
- [ ] Help
- [ ] Import
- [ ] Export
- [ ] Macro
- [ ] Tag
- [ ] Untag

### PR DESCRIPTION
**Pull request by JoeyChenSmart**
[What I've done](https://www.youtube.com/watch?v=8sgycukafqQ) in this PR
- [x] Change the writeup on logic component
- [x] Added a link to the [PR](https://github.com/AY2021S1-CS2103T-W17-3/tp/pull/39) for more details on the new logic architecture.
- [x] Replace diagrams in logic (1 class diagram + 1 sequence diagram)
- [x] Made a short writeup on macros
- [x] Add class and sequence diagram for macros

### CODE REVIEW

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 1]
```diff
@@ -16,34 +16,38 @@ end box
...
 
-DeleteCommandParser --> AddressBookParser : d
-deactivate DeleteCommandParser
-'Hidden arrow to position the destroy marker below the end of the activation bar.
-DeleteCommandParser -[hidden]-> AddressBookParser
-destroy DeleteCommandParser
+|||
+
+PrimitiveCommandParser --> DeleteCommand : getParameterSet
+activate DeleteCommand
```
Comment (chewypiano) [inline docs/diagrams/DeleteSequenceDiagram.puml]: Should this be -> instead of --> ?

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 2]
```diff
@@ -16,34 +16,38 @@ end box
...
+DeleteCommand --> PrimitiveCommandParser
+deactivate DeleteCommand
 
-AddressBookParser --> LogicManager : d
-deactivate AddressBookParser
+|||
+
+PrimitiveCommandParser --> PrimitiveCommandParser : provideValuesToParameterSet
+activate PrimitiveCommandParser
```
Comment (chewypiano) [inline docs/diagrams/DeleteSequenceDiagram.puml]: Similar to previous comment, shouldn't this be -> instead of -->?

**Review by chewypiano — COMMENTED** [review]
Looks very good, just the most minor nits.

File: docs/diagrams/DeleteSequenceDiagram.puml (hunk 1 above)
Comment (JoeyChenSmart) [reply docs/diagrams/DeleteSequenceDiagram.puml]: yup you're probably right. I mustve copy pasted wrongly somewhere

**Review by chewypiano — APPROVED** [review]
LGTM

**Pull request merged**